gravitee-apim-console-webui/src/management/settings/api-quality-rules-ng/api-quality-rules-ng.component.ts

```diff
@@ -0,0 +1,248 @@
+import { catchError, filter, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { GIO_DIALOG_WIDTH, GioConfirmDialogComponent, GioConfirmDialogData } from '@gravitee/ui-particles-angular';
+import { MatDialog } from '@angular/material/dialog';
+
+import { ApiQualityRulesNgAddDialogComponent } from './api-quality-rules-ng-add-dialog/api-quality-rules-ng-add-dialog.component';
+
+import { Constants, EnvSettings } from '../../../entities/Constants';
+import { PortalSettingsService } from '../../../services-ngx/portal-settings.service';
+import { SnackBarService } from '../../../services-ngx/snack-bar.service';
+import { EnvironmentSettingsService } from '../../../services-ngx/environment-settings.service';
+import { QualityRuleService } from '../../../services-ngx/quality-rule.service';
+import { QualityRule } from '../../../entities/qualityRule';
+import { GioTableWrapperFilters } from '../../../shared/components/gio-table-wrapper/gio-table-wrapper.component';
+import { gioTableFilterCollection } from '../../../shared/components/gio-table-wrapper/gio-table-wrapper.util';
+
+export interface ApiQualityRulesDialogData {
+  weight: number;
+  name: string;
+  description: string;
+  id?: string;
+}
+
+@Component({
+  selector: 'api-quality-rules-ng',
+  templateUrl: './api-quality-rules-ng.component.html',
+  styleUrls: ['./api-quality-rules-ng.component.scss'],
+})
+export class ApiQualityRulesNgComponent implements OnInit {
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+  settings: EnvSettings;
+  apiQualityRulesForm: UntypedFormGroup;
+  public formInitialValues: unknown;
+  public qualityRulesListTable: QualityRule[] = [];
+  public displayedColumns = ['quality', 'description', 'weight', 'actions'];
+  public qualityRulesUnpaginatedLength = 0;
+  public filteredQualityRulesSettingsTable: QualityRule[] = [];
+  public isLoadingData = true;
+  public initialFilters: GioTableWrapperFilters = {
+    pagination: {
+      size: 10,
+      index: 1,
+    },
+    searchTerm: '',
+  };
+
+  constructor(
+    @Inject('Constants') private readonly constants: Constants,
+    private readonly portalSettingsService: PortalSettingsService,
+    private readonly snackBarService: SnackBarService,
+    private readonly environmentSettingsService: EnvironmentSettingsService,
+    private readonly qualityRuleService: QualityRuleService,
+    private readonly matDialog: MatDialog,
+  ) {}
+
+  public ngOnInit() {
+    this.isLoadingData = true;
+    this.settings = this.constants.env.settings;
+    this.qualityRuleService
+      .list()
+      .pipe(
+        tap((qualityRulesList) => {
+          this.qualityRulesListTable = qualityRulesList;
+          this.onPropertiesFiltersChanged(this.initialFilters);
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe(() => {
+        this.isLoadingData = false;
+      });
+
+    this.apiQualityRulesForm = new UntypedFormGroup({
```

Comment:
What do you think about typing this form now that we can ? 
We can check with other's APIM member  but we could force ourselves to type forms on all the new APIM pages. cc @jourdiw @Okhelifi  & @gravitee-io/apim

Reply:
I think that could be a nice enhancement!

Reply:
On our side we try to always use typed form on new components. I would even say that we should try to type the forms we we refactor an existing component. It would be nice to do it to migrate all the console to typed forms step by step.

Reply:
Done